### PR TITLE
ID-263, ID-263: Enable sts and cache control headers.

### PR DESCRIPTION
**Pull request by Ghost-in-a-Jar**
This pr addresses both https://broadworkbench.atlassian.net/browse/ID-263 and https://broadworkbench.atlassian.net/browse/ID-262

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary. 
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by tlangs — APPROVED** [review]

**Review by Shakespeared — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -93,3 +93,16 @@ def add_nosniff_content_type_header(response):
     response.headers["X-Content-Type-Options"] = "nosniff"
     response.headers["X-Frame-Options"] = "deny"
     return response
+
+@app.after_request
```
Comment (andy7i) [inline main.py:97]: (asking not knowing a lot about flask) -- it seems weird to add a header after the response is sent. Or does `after_request` actually run before the response is sent?

Comment (andy7i) [reply main.py:97]: found the answer. it's run after the request is processed but before the response is sent...
flask request response timing
https://instructobit.com/tutorial/111/Python-Flask%3A-running-code-before-and-after-every-request

File: main.py [hunk 2]
```diff
@@ -93,3 +93,16 @@ def add_nosniff_content_type_header(response):
...
+def add_strict_transport_security_header(response):
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+    # recommended settings by mozilla, max-age is 2 years in seconds
```
Comment (andy7i) [inline main.py:100]: I took a look through the link (which links to hstspreload.org) and there's something about:
> Once you're confident that there will be no more issues, increase the max-age to 2 years and submit your site to the preload list

from https://hstspreload.org/

I'm not very familiar with the preload list but is it something we should be concerned about?

**Review by andy7i — COMMENTED** [review]
looks good overall! thanks for adding in the helpful comments too. 

I have a small concern about the hsts preload stuff. If that's fine I can give a thumb. 

Also, I think we can add ID-262 to the title (since this PR addresses 2 tickets)

File: main.py (hunk 2 above)
Comment (Ghost-in-a-Jar) [reply main.py:100]: @andy7i I think this is more of a devops thing where they would register the terra ui dns record with that hstspreload list

Comment (Ghost-in-a-Jar) [reply main.py:100]: it seems to be out of scope of this ticket and pentest requirements

Comment (andy7i) [reply main.py:100]: one of my concerns was that 2 years might not be the right amount of time to put here. Mozilla says:

> **If** a max-age of 1 year is acceptable for a domain, however, two years is the recommended value as explained on https://hstspreload.org/.

emphasis on `if`. Do we know if 1 year is acceptable?

if you look at https://hstspreload.org/#deployment-recommendations the deployment recommendations slowly ramp up the max-age (starting with 5 minutes and 1 week intervals) .

**Pull request merged**